src/client/Client.cc

```diff
@@ -11584,8 +11584,12 @@ int Client::_do_setxattr(Inode *in, const char *name, const void *value,
 {
 
   int xattr_flags = 0;
-  if (!value)
-    xattr_flags |= CEPH_XATTR_REMOVE;
+  // If the size is set to 0, set the value to an empty string
+  if (size == 0) {
+    value = "";
+  } else if (value == NULL) {
+      return -EINVAL;
```

Comment:
@jtlayton did I misremember what you said? I thought Linux will treat a NULL value as `""`?

Reply:
Only when size is also 0.

Reply:
Only when size is 0.